### DISCUSSION TITLE
UGWA/gunn.app contingency plans

### DISCUSSION THREAD

**Discussion by SheepTester** [Orbiit/gunn-web-app#200] [Ideas]
Since UGWA is mostly maintained by students of the class of 2021, who are graduating this year, what should happen to UGWA and the gunn.app domain in future school years?

There now exist several successors to UGWA: [WATT](https://gunnwatt.web.app/) and [Campus Central](https://campuscentral.repl.co/)

Informally, "gunn.app" stems from marketing and often is used to refer to UGWA. However, for clarity, because there are multiple independent properties that can be transferred, we should use the following convention:

- **UGWA** refers to this repository and its corresponding GitHub Pages site (the app itself)
- **gunn.app** refers to the *domain*, which is owned by @anematode, that currently redirects to UGWA
- **Orbiit** refers to the GitHub organization
- **OOPS** refers to the propaganda department that advertises UGWA and also refers collectively to decisions made regarding UGWA etc.

I will list existing plans/ideas here, and others can suggest their own ideas here as well

**Comment by SheepTester** [comment]
I currently plan to add @chop0, @icedTet, and @FrozenMango747 as members to **Orbiit**—this organization—so they can update UGWA if they so please.

**Reply by ky28059** [reply]
What about me 😢

**Reply by icedTet** [reply]
> What about me 😢

~~you use DJS not **eris**; disqualified~~

**Reply by ky28059** [reply]
I am technically an UGWA contributor

**Reply by icedTet** [reply]
> I am technically an UGWA contributor

~~But you use d.js tho.~~

**Reply by SheepTester** [reply]
I can add you as well, Kevin, if you want

**Comment by SheepTester** [comment]
I believe someone suggested an idea to put the successor apps up to a vote some time after the end of the school year, and **gunn.app** will then redirect to the winning app.

**Comment by SheepTester** [comment]
To avoid disrupting the common students next year, who I doubt will stay up to date with any OOPS shenanigans, I plan on updating **UGWA** with minimal changes for the new year (i.e. changing school year start/end dates, etc.), and then cease updating UGWA from that point onwards. This way, if they stay under a rock over the summer and instinctively open UGWA on the first day of school, they can still get through their day without having to worry about switching apps in the morning.

**Comment by SheepTester** [comment]
The current plan for **UGWA** is to show something akin to the following message over the app starting the last day of this school year (apps should be randomized to avoid bias):

> [Graduation cap icon]
>
> ### Like the class of 2021, UGWA will graduate this year.
>
> To stay up to date, we recommend switching to an app made by current Gunn students:
>
> WATT | Campus Central
> --- | ---
> Also supports Schoology integration and some other cool features (give a rundown here) | Also supports Schoology integration and some other cool features (give a rundown here)
>
> (Did you make your own app? [Make an issue](#))
>
> Or [continue using UGWA](#)
>
> - [ ] Don't show again for like a month

If the app starts during the school day, it shouldn't appear until after school to avoid disruptions. Also, the design should be easily expandable as new apps are made or old ones are abandoned

**Reply by SheepTester** [reply]
This could also track the number of times each link is clicked, which can be used to determine which app to redirect to for **gunn.app** (per org. teams)

**Reply by icedTet** [reply]
Why not just have a link portal and let the user choose then save their choice in localStorage so they only need to do it once?

**Reply by SheepTester** [reply]
Done!

<img src="https://user-images.githubusercontent.com/22133785/120715187-1bb70c00-c479-11eb-8989-4da2323a151f.png" width="300" alt="Screenshot of UGWA's graduation modal">
